code/modules/tgui/tgui.dm

```diff
@@ -305,8 +309,13 @@
 	switch(type)
 		if("ready")
 			// Send a full update when the user manually refreshes the UI
-			if (initialized && COOLDOWN_FINISHED(src, refresh_cooldown))
-				send_full_update()
+			if (initialized)
+				if(!COOLDOWN_FINISHED(src, refresh_cooldown))
+					if(!refreshing)
+						addtimer(CALLBACK(src, .proc/send_full_update), TGUI_REFRESH_FULL_UPDATE_COOLDOWN)
```

Comment:
make sure this is a unique timer so it can't be queued or spammed

Reply:
indeed - that's what i'm using the ```refreshing``` var for. the only way to flick off refreshing is for send_full_update to fire on the window.

if there is a better way to do this, let me know, i'm not very familiar with timers

Reply:
It's a confusing design, but I don't know what to suggest.

Reply:
i tried writing it differently. dont know how/why i let it get to 3 nested ifs